app/async_database.py

```diff
@@ -0,0 +1,162 @@
+        """Disable the game in a channel.
+
+        Arguments:
+        ---------
+        channel (discord.TextChannel): The channel that the game is to be disabled in
+
+        """
+        await self.connection.execute("DELETE FROM Guilds WHERE id=? AND channel=?", (channel.guild.id, channel.id))
+        await self.connection.commit()
+
+    async def get_channels(self, guild: discord.Guild) -> list[discord.TextChannel]:
+        """Get all the channels that the game is in enabled in for a guild.
+
+        Arguments:
+        ---------
+        guild (discord.Guild): The guild to check all enabled profiles in
+
+        """
+        async with (
+            self.connection.execute("SELECT * FROM Guilds WHERE guild_id = ?", (guild.id)) as cursor,
+        ):
+            return [discord.Client.get_channel(row[1]) for row in cursor]
+
+    async def get_active_profiles(self, guild: discord.Guild) -> list[UserProfile]:
+        """Get all the profiles that the game is in enabled in.
+
+        Arguments:
+        ---------
+        guild (discord.Guild): The guild to check all enabled profiles in
+
+        """
+        async with (
+            self.connection.execute("SELECT * FROM Users WHERE guild_id = ?", (guild.id)) as cursor,
+        ):
+            return [UserProfile(coins=row[3], cps=row[2]) for row in cursor]
+
+    async def add_profile(self, guild: discord.Guild, user: discord.Member) -> None:
+        """Add a profile to a specific guild.
+
+        Arguments:
+        ---------
+        guild (discord.Guild): The guild that the user is in
+        user (discord.Member): This user whose profile is to be added
+
+        """
+        await self.connection.execute(
+            "INSERT INTO Users(user_id, guild_id, cps, coins) VALUES (?,?,?,?)",
+            (user.id, guild.id, UserProfile().cps, UserProfile().coins),
+        )
+        await self.connection.commit()
+
+    async def remove_profile(self, guild: discord.Guild, user: discord.Member) -> None:
+        """Remove a profile from a specific guild.
+
+        Arguments:
+        ---------
+        guild (discord.Guild): The guild that the user is in
+        user (discord.Member): This user whose profile is to be removed
+
+        """
+        await self.connection.execute("DELETE FROM Users WHERE user_id=? AND guild_id=?", (user.id, guild.id))
+        await self.connection.commit()
+
+    async def get_profile(self, guild: discord.Guild, user: discord.Member) -> UserProfile | None:
+        """Get a profile from a specific guild, if the user object does not have the guild already attached to it.
+
+        Arguments:
+        ---------
+        guild (discord.Guild): The guild that will be checked
+        user (discord.Member): The user whose profile that will be returned
+
+        """
+        async with (
+            self.connection.execute(
+                "SELECT * FROM Users WHERE guild_id = ? AND user_id = ?", (guild.id, user.id)
```

Comment:
I would prefer selecting specific column names here so that we can change the DB schemas (add new columns at the start) without issues.